core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -990,11 +990,19 @@ private[spark] object Utils extends Logging {
     for ((key, value) <- extraEnvironment) {
       environment.put(key, value)
     }
+
+    // When testing, expose the parent class path to the child. This is processed by
+    // compute-classpath.{cmd,sh} and makes all needed jars available to child processes
+    // when the assembly is built with the "*-provided" profiles enabled.
+    if (sys.props.contains("spark.testing")) {
+      environment.put("SPARK_DIST_CLASSPATH", sys.props("java.class.path"))
```

Comment:
Even if it is still a problem, my question was, can we just set `SPARK_DIST_CLASSPATH` in the child to the value of `SPARK_DIST_CLASSPATH` in the parent? This is instead of setting it to the value of `java.class.path`.